### DISCUSSION TITLE
Vector-Jacobian Product

### DISCUSSION THREAD

**Discussion by pharringtonp19** [google/flax#1434] [Q&A] [answered]
In the Vector-Jacobain product section of [The Jax for the Impatient](https://flax.readthedocs.io/en/latest/notebooks/jax_for_the_impatient.html) tutorial, the authors explain how to efficiently compute the gradient of a function composed of two functions.

They note that the outer function, `\phi`, is  an element of the dual space of `R^m`. I am curious about why they make this note. What changes if we don't impose that `\phi` is an element of the dual space?

**Comment by andsteing** [comment]
@BertrandRdp who was originally writing this tutorial.

**Comment by BertrandRdp** [comment, marked as answer]
Hi! I don't remember exactly what I was thinking, but at the time I was reflecting on why you would be interested in computing those quantities (from a mathematical perspective). However rereading now seems that as I was thinking about the space where the gradient lives (dual space), and I somehow mixed the original function and its gradient.

Hence `\phi` is a scalar valued function (not a linear form), its differential is a linear form (hence the dual space). In the case where `\phi` is a linear form, the only thing that changes is that the gradient is a constant vector, which is just a special case.

I'll fix this, thanks for catching this :)

**Reply by andsteing** [reply]
Thanks for fixing this in #1442, Bertrand!

**Reply by pharringtonp19** [reply]
@BertrandRdp Thanks for the explanation!
